PantheonCMD/pcbuild.py

```diff
@@ -124,40 +111,150 @@ def coalesce_document(main_file, attributes=None, depth=0, top_level=True):
 
     # Open the file, iterate over lines
     if os.path.exists(main_file):
+
         with open(main_file) as input_file:
+
             # Iterate over content
             for line in input_file:
-                # Process comment block start and finish
-                if line.strip().startswith('////'):
+
+                # CONDITIONS
+
+                # Line matches the end of a condition_block
+                matches = re.match(r'^endif::(.*?)\[\]', line)
+
+                if matches:
+                    if matches.group(1) == '':
+                        condition_block = False
+                    else:
+                        if matches.group(1) == conditions_set:
+                            conditions_set = ''
+                            condition_block = False
+                    continue
```

Comment:
I'm a bit confused about this part:

in both if and else statements matches.group(1) == '', since conditions_set = ''. so why not:

```suggestion
                if matches:
                    if matches.group(1) == conditions_set:
                            conditions_set = ''
                            condition_block = False
                    continue
```

I might be wring here though

Reply:
Hey @Levi-Leah - this is attempting to address two use cases.

First - if there's a blank endif statement, it doesn't matter what the conditions are, we end the block.
Second - if there's an endif statement that includes some conditions (for example endiff:[condition1,condition2]), only end the block if the conditions match the ones that started the block.